src/admm/polar_kernel.jl

```diff
@@ -189,31 +189,27 @@ function polar_kernel_cpu(n::Int, nline::Int, line_start::Int, scale::Float64,
     return 0, avg_minor_it / nline
 end
 
-function polar_kernel_two_level(
-    n::Int, nline::Int, line_start::Int, bus_start::Int, scale::Float64,
-    u::CuDeviceArray{Float64}, xbar::CuDeviceArray{Float64},
-    z::CuDeviceArray{Float64}, l::CuDeviceArray{Float64}, rho::CuDeviceArray{Float64},
-    shift_lines::Int, param::CuDeviceArray{Float64},
-    _YffR::CuDeviceArray{Float64}, _YffI::CuDeviceArray{Float64},
-    _YftR::CuDeviceArray{Float64}, _YftI::CuDeviceArray{Float64},
-    _YttR::CuDeviceArray{Float64}, _YttI::CuDeviceArray{Float64},
-    _YtfR::CuDeviceArray{Float64}, _YtfI::CuDeviceArray{Float64},
-    frBound::CuDeviceArray{Float64}, toBound::CuDeviceArray{Float64},
-    brBusIdx::CuDeviceArray{Int}
+@kernel function polar_kernel_two_level(
+    @Const(n::Int), nline::Int, line_start::Int, bus_start::Int, scale::Float64,
+    u, xbar,
+    z, l, rho,
+    shift_lines::Int, param,
+    _YffR, _YffI,
+    _YftR, _YftI,
+    _YttR, _YttI,
+    _YtfR, _YtfI,
+    frBound, toBound,
+    brBusIdx
 )
-    I = blockIdx().x
-    id_line = I + shift_lines
+    I = @index(Group, Linear)
+    J = @index(Local, Linear)
 
-    x = @cuDynamicSharedMem(Float64, n)
-    xl = @cuDynamicSharedMem(Float64, n, n*sizeof(Float64))
-    xu = @cuDynamicSharedMem(Float64, n, (2*n)*sizeof(Float64))
+    x = @localmem Float64 (4,)
```

Comment:
I see we define `n` as `Const`, and then use `4` hardcoded here. Is there any performance related issue?